llvm/lib/Transforms/Vectorize/VPlanVerifier.cpp

```diff
@@ -156,7 +156,8 @@ bool VPlanVerifier::verifyEVLRecipe(const VPInstruction &EVL) const {
              .Case<VPScalarCastRecipe>(
                  [&](const VPScalarCastRecipe *S) { return true; })
              .Case<VPInstruction>([&](const VPInstruction *I) {
-               if (I->getOpcode() != Instruction::Add) {
+               if ((I->getOpcode() != Instruction::Add) &&
+                   (I->getOpcode() != Instruction::Mul)) {
                  errs()
                      << "EVL is used as an operand in non-VPInstruction::Add\n";
```

Comment:
Need Update?

Reply:
The step increment `mul(step, evl)` is added to the plan, so we need to relax the constraint.

Reply:
There is a misunderstanding. I mean update assert message.
```
errs() << "EVL is used as an operand in non-VPInstruction::Add\n";
```